src/Symfony/Component/Form/Extension/Core/Type/TimeType.php

```diff
@@ -49,6 +51,18 @@ public function buildForm(FormBuilderInterface $builder, array $options)
 
         if ('single_text' === $options['widget']) {
             $builder->addViewTransformer(new DateTimeToStringTransformer($options['model_timezone'], $options['view_timezone'], $format));
+
+            // handle seconds ignored by user's browser when with_seconds enabled and seconds is 00
+            if ($options['with_seconds']) {
+                $builder->addEventListener(FormEvents::PRE_SUBMIT, function (FormEvent $e) {
+                    if ($data = $e->getData()) {
+                        if (preg_match('/^\d{2}:\d{2}$/', $data)) {
```

Comment:
Implemented suggestions, has not simplified the condition 'if ($ options [' with_seconds '])' to avoid adding an not necessary listener